parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestEncryption.java

```diff
@@ -0,0 +1,215 @@
+        .withAADPrefix(AADPrefix)
+        .withEncryptedColumns(columnPropertiesMap)
+        .build();
+    encryptionPropertiesList[4] = encryptionProperties;
+    decryptionPropertiesList[4] = decryptionProperties; // Same decryption properties
+
+
+    MessageType schema = parseMessageType(
+        "message test { "
+            + "required binary binary_field; "
+            + "required int32 int32_field; "
+            + "required int64 int64_field; "
+            + "required boolean boolean_field; "
+            + "required float float_field; "
+            + "required double double_field; "
+            + "required fixed_len_byte_array(3) flba_field; "
+            + "required int96 int96_field; "
+            + "} ");
+    GroupWriteSupport.setSchema(schema, conf);
+    SimpleGroupFactory f = new SimpleGroupFactory(schema);
+
+    for (int encryptionMode = 0; encryptionMode < numberOfEncryptionModes; encryptionMode++) {
+      System.out.println("MODE: "+encryptionMode);
+      
+      Path file = new Path(root, "m_" + encryptionMode + ".parquet.encrypted");
+      ParquetWriter<Group> writer = new ParquetWriter<Group>(
+          file,
+          new GroupWriteSupport(),
+          UNCOMPRESSED, 1024, 1024, 512, true, false, ParquetWriter.DEFAULT_WRITER_VERSION, conf, 
+          encryptionPropertiesList[encryptionMode]);
+      for (int i = 0; i < 1000; i++) {
+        writer.write(
+            f.newGroup()
+            .append("binary_field", "test" + i)
+            .append("int32_field", 32)
+            .append("int64_field", 64l)
+            .append("boolean_field", true)
+            .append("float_field", 1.0f)
+            .append("double_field", 2.0d)
+            .append("flba_field", "foo")
+            .append("int96_field", Binary.fromConstantByteArray(new byte[12])));
+      }
+      writer.close();
+
+      FileDecryptionProperties fileDecryptionProperties = decryptionPropertiesList[encryptionMode];
+      ParquetReader<Group> reader = ParquetReader.builder(new GroupReadSupport(), file)
+          .withDecryption(fileDecryptionProperties).withConf(conf).build();
+      for (int i = 0; i < 1000; i++) {
+        Group group = null;
+        group= reader.read();
+        assertEquals("test" + i, group.getBinary("binary_field", 0).toStringUsingUTF8());
+        assertEquals(32, group.getInteger("int32_field", 0));
+        assertEquals(64l, group.getLong("int64_field", 0));
+        assertEquals(true, group.getBoolean("boolean_field", 0));
+        assertEquals(1.0f, group.getFloat("float_field", 0), 0.001);
+        assertEquals(2.0d, group.getDouble("double_field", 0), 0.001);
+        assertEquals("foo", group.getBinary("flba_field", 0).toStringUsingUTF8());
+        assertEquals(Binary.fromConstantByteArray(new byte[12]),
+            group.getInt96("int96_field",0));
+      }
+      reader.close();
+    }
+    enforceEmptyDir(conf, root);
+  }
+
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
```

Comment:
Yep, that's one good way I was talking about but you should use `temp` instead of hardcoding the directory at the beginning.